runtime/src/bank.rs

```diff
@@ -2028,13 +2043,16 @@ impl Bank {
                 // flush the Stakes cache
                 account.lamports = 0;
                 self.store_account(&program_id, &account);
+                None
```

Comment:
add a test justifying this new behavior

Reply:
Improved existing: https://github.com/solana-labs/solana/pull/13884/files#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cR8633
Also wrote a new test case specifically for this: https://github.com/solana-labs/solana/pull/13884/files#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cR10141